sim-lib/src/lib.rs

```diff
@@ -815,32 +815,37 @@ async fn produce_random_events<N: NetworkGenerator, A: PaymentGenerator + Displa
             // Wait until our time to next payment has elapsed then execute a random amount payment to a random
             // destination.
             _ = time::sleep(wait) => {
-                let destination = network_generator.lock().await.sample_node_by_capacity(source.pubkey);
+                let (destination, capacity) = network_generator.lock().await.sample_node_by_capacity(source.pubkey);
```

Comment:
I would have thought we want to do this validation much higher up? 

Like nodes that don't have enough capacity, if a node doesn't support keysend when we start running it never will so we should just exclude it from the start (and log a warning so the user knows why).

If we do this validation here, we could also _theoretically_ end up effectively running with one node (in the case of two nodes and one doesn't have keysend) and then just run without ever sending a payment.

Reply:
We can run this much higher up, as long as we are ok with compromising with `--accept-keysend` being enabled for both source and destination nodes. Otherwise, we cannot know if `keysend` receiving is really going to be necessary until a node is picked as a payment destination, and that happens on the fly.

Reply:
> compromising with --accept-keysend being enabled for both source and destination nodes.

In the case of random activity I think that's okay because every sending node can reasonably be expected to also receive?

Reply:
Fair enough. https://github.com/bitcoin-dev-project/sim-ln/pull/103/commits/98c15d076e28c96416b6151ef9ac6fd68cef2715 should have fixed that